admin/class-aioseop-notices.php

```diff
@@ -0,0 +1,710 @@
+		public $notices = array();
+
+		/**
+		 * List of notice slugs that are currently active.
+		 * NOTE: Amount is reduced by 1 second in order to display at exactly X amount of time.
+		 *
+		 * @since 3.0
+		 * @access public
+		 *
+		 * @var array $active_notices {
+		 *     @type string|int $slug => $display_time Contains the current active notices
+		 *                                             that are scheduled to be displayed.
+		 * }
+		 */
+		public $active_notices = array();
+
+		/**
+		 * The default dismiss time. An anti-nag setting.
+		 *
+		 * @var int $default_dismiss_delay
+		 */
+		private $default_dismiss_delay = 180;
+
+		/**
+		 * List of Screens used in AIOSEOP.
+		 *
+		 * @since 3.0
+		 *
+		 * @var array $aioseop_screens {
+		 *     @type string Screen ID.
+		 * }
+		 */
+		private $aioseop_screens = array(
+			'toplevel_page_' . AIOSEOP_PLUGIN_DIRNAME . '/aioseop_class',
+			'all-in-one-seo_page_' . AIOSEOP_PLUGIN_DIRNAME . '/modules/aioseop_performance',
+			'all-in-one-seo_page_' . AIOSEOP_PLUGIN_DIRNAME . '/modules/aioseop_sitemap',
+			'all-in-one-seo_page_aiosp_opengraph',
+			'all-in-one-seo_page_aiosp_robots_generator',
+			'all-in-one-seo_page_' . AIOSEOP_PLUGIN_DIRNAME . '/modules/aioseop_robots',
+			'all-in-one-seo_page_' . AIOSEOP_PLUGIN_DIRNAME . '/modules/aioseop_file_editor',
+			'all-in-one-seo_page_' . AIOSEOP_PLUGIN_DIRNAME . '/modules/aioseop_importer_exporter',
+			'all-in-one-seo_page_' . AIOSEOP_PLUGIN_DIRNAME . '/modules/aioseop_bad_robots',
+			'all-in-one-seo_page_' . AIOSEOP_PLUGIN_DIRNAME . '/modules/aioseop_feature_manager',
+		);
+
+		/**
+		 * __constructor.
+		 *
+		 * @since 3.0
+		 */
+		public function __construct() {
+			$this->_requires();
+			if ( current_user_can( 'aiosp_manage_seo' ) ) {
+				$this->obj_load_options();
+
+				add_action( 'admin_init', array( $this, 'init' ) );
+				add_action( 'current_screen', array( $this, 'admin_screen' ) );
+			}
+		}
+
+		/**
+		 * _Requires
+		 *
+		 * Additional files required.
+		 *
+		 * @since 3.0
+		 */
+		private function _requires() {
+			require_once AIOSEOP_PLUGIN_DIR . 'admin/functions-notice.php';
+		}
+
+		/**
+		 * Early operations required by the plugin.
+		 *
+		 * AJAX requires being added early before screens have been loaded.
+		 *
+		 * @since 3.0
+		 */
+		public function init() {
+			add_action( 'wp_ajax_aioseop_notice', array( $this, 'ajax_notice_action' ) );
+		}
+
+		/**
+		 * Setup/Init Admin Screen
+		 *
+		 * Adds the initial actions to WP based on the Admin Screen being loaded.
+		 * The AIOSEOP and Other Screens have separate methods that are used, and
+		 * additional screens can be made exclusive/unique.
+		 *
+		 * @since 3.0
+		 *
+		 * @param WP_Screen $current_screen The current screen object being loaded.
+		 */
+		public function admin_screen( $current_screen ) {
+			$this->deregister_scripts();
+			if ( in_array( $current_screen->id, $this->aioseop_screens, true ) && isset( $current_screen->id ) ) {
```

Comment:
@EkoJR the conditions should be swapped. check for isset first

Reply:
Fixed